File: 4.5.pas
program SerialNumbersAndPowerCapacities;
var
  N, i, maxIndex: integer;
  power, sum, maxPower: real;
begin
  write('Введите количество предприятий N: ');
  readln(N);
  sum := 0;
  maxPower := -1;
  maxIndex := 0;
  for i := 1 to N do
  begin
    write('Введите мощность предприятия ', i, ': ');
    readln(power);
    sum := sum + power;
    if power > maxPower then
    begin
      maxPower := power;
      maxIndex := i;
    end;
  end;
  writeln('Общая мощность: ', sum:0:2);
  writeln('Средняя мощность: ', (sum / N):0:2);
  writeln('Номер предприятия с наибольшей мощностью: ', maxIndex);
end.

File: 4.5.py
N = int(input("Введите количество предприятий: "))
sum_power = 0
max_power = -1
max_index = 0
for i in range(1, N + 1):
    power = float(input(f"Введите мощность предприятия {i}: "))
    sum_power += power
    if power > max_power:
        max_power = power
        max_index = i
print(f"Общая мощность: {sum_power:.2f}")
print(f"Средняя мощность: {sum_power / N:.2f}")
print(f"Номер предприятия с наибольшей мощностью: {max_index}")